contracts/protocol/integration/amm/ArrakisUniswapV3AmmAdapter.sol

```diff
@@ -0,0 +1,277 @@
+
+        (uint256 amountAMin, uint256 amountBMin, uint256 liquidityExpectedFromSuppliedTokens) = arrakisVaultPool.getMintAmounts(maxTokensInA, maxTokensInB);
+        
+        require(
+            minLiquidity <= liquidityExpectedFromSuppliedTokens,
+            "_minLiquidity is too high for input token limit"
+        );
+
+        target = router;
+        value = 0;
+        data = abi.encodeWithSignature( 
+            ADD_LIQUIDITY,
+            arrakisVaultPool,
+            maxTokensInA,
+            maxTokensInB,
+            amountAMin,
+            amountBMin,
+            setToken
+        );
+    }
+
+    /**
+     * Return calldata for the add liquidity call for a single asset
+     */
+    function getProvideLiquiditySingleAssetCalldata(
+        address /*_setToken*/,
+        address /*_pool*/,
+        address /*_component*/,
+        uint256 /*_maxTokenIn*/,
+        uint256 /*_minLiquidity*/
+    )
+        external
+        pure
+        override
+        returns (address /*target*/, uint256 /*value*/, bytes memory /*data*/)
+    {
+        revert("Arrakis single asset addition is not supported");
+    }
+
+    /**
+     * Return calldata for the remove liquidity call
+     *
+     * @param  _setToken                Address of the SetToken
+     * @param  _pool                    Address of liquidity token
+     * @param  _components              Token address array required to remove liquidity
+     * @param  _minTokensOut            AmountsOut minimum to remove liquidity
+     * @param  _liquidity               Liquidity amount to remove
+     */
+    function getRemoveLiquidityCalldata(
+        address _setToken,
+        address _pool,
+        address[] calldata _components,
+        uint256[] calldata _minTokensOut,
+        uint256 _liquidity
+    )
+        external
+        view
+        override
+        returns (address target, uint256 value, bytes memory data)
+    {   
+        address setToken = _setToken;
+        address[] memory components = _components;
+        uint256[] memory minTokensOut = _minTokensOut;
+        uint256 liquidity = _liquidity;
+        IArrakisVaultV1 arrakisVaultPool = IArrakisVaultV1(_pool);
+
+        // Make sure that only up to the amount of liquidity tokens owned by the Set Token are redeemed
+        uint256 setTokenLiquidityBalance = arrakisVaultPool.balanceOf(setToken);
+        require(liquidity <= setTokenLiquidityBalance, "_liquidity must be <= to current balance");
+
+        // Checks for minTokensOut
+        require(minTokensOut[0] > 0 && minTokensOut[1] > 0, "Minimum quantity must be nonzero");
+
+        // Sort the amount in order of tokens stored in Arrakis Pool
+        (uint256 minTokensOutA, uint256 minTokensOutB) = _getOrderedAmount(components[0], components[1], minTokensOut[0], minTokensOut[1]);
+
+        target = router;
+        value = 0;
+        data = abi.encodeWithSignature(
```

Comment:
I think we could do something like:
`abi.encodeWithSelector(IArrakis.removeLiquidity ... ` here no ? 

Advantages:
1. No need of  signature string constant
2. Less error prone
3. Maybe more efficient if the selector is precomputed at compile time